pkg/manifest/manifest.go

```diff
@@ -0,0 +1,118 @@
+package manifest
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// Manifest stores Kubernetes object in Raw from a file.
+// It stores the GroupVersionKind for the manifest.
+type Manifest struct {
```

Comment:
what are the invariants? Raw and Obj are always in sync and set?

Reply:
They are always in sync with it's current use by CVO and my planned use by oc. The struct only gets set here by this call to ParseManifests in CVO:

https://github.com/jottofar/cluster-version-operator/blob/732ae625e991bf1447370b537caf9bcc22c850dc/pkg/payload/payload.go#L147

Also, both manifest.ManifestsFromFiles and manifest.UnmarshalJSON set both

Reply:
please document that. It's central for the type.

Reply:
documented